### ISSUE TITLE
Issues with network instrumentation after bumping to 2.2.0

### ISSUE THREAD

**Issue by dcacenabes** [DataDog/dd-sdk-swift-testing#76]
### UI Test execution increase after bumping to 2.2.0

TL;DR
Starting in version 2.2.0, network instrumentation adds between 2-3 minutes delay to each test execution.

We were using an older version of the SDK in our apps. We just migrated to the latest version and noticed that UI tests went from ~15 seconds each to ~190 seconds.
We have been testing different versions of the lib until we pinned it down to [2.2.0](https://github.com/DataDog/dd-sdk-swift-testing/releases/tag/2.2.0)

The behavior that we are observing is that the test takes a very long time to start. Once it finally starts, it has a normal execution time.

We have only noted this for UI tests, not unit tests. After researching, I think this is related to the fact that for UI tests we do some real network requests to setup stubs in a mock server. We see this in the logs:

```
2023-06-12 10:48:05.175739+0200 XXXXX-Runner[6034:5463357] [Client] Updating selectors after delegate addition failed with: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named com.apple.commcenter.coretelephony.xpc was invalidated: failed at lookup with error 3 - No such process." UserInfo={NSDebugDescription=The connection to service named com.apple.commcenter.coretelephony.xpc was invalidated: failed at lookup with error 3 - No such process.}
2023-06-12 10:49:58.153925+0200 XXXXX-Runner[6034:5464533] [connection] nw_connection_add_timestamp_locked_on_nw_queue [C3] Hit maximum timestamp count, will start dropping events
```

There are multiple occurrences of the error `com.apple.commcenter.coretelephony.xpc was invalidated`, and then after `Hit maximum timestamp count, will start dropping events` the test starts running as expected.

Setting `DD_DISABLE_NETWORK_INSTRUMENTATION` to true **fixes the issue**, but I wonder what has changed in the library that makes this a problem all of a sudden.

This is not a problem for us since we don't rely on network instrumentation for the tests, but you might want to take a look at the issue.

**Comment by ypopovych** [comment]
Hello,

We released a 2.5.0 version of the SDK which uses updated OpenTelemetry SDK. It should fix that problem.

Thanks.
